### PR TITLE
Arf 105 create pull request api

### LINKED ISSUE #105: Support the Create Pull Request API
https://developer.github.com/v3/pulls/#create-a-pull-request

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #105  Create PullRequest Api integration

### CODE REVIEW

File: github4s/jvm/src/test/scala/github4s/utils/TestUtils.scala [hunk 1]
```diff
@@ -111,6 +111,18 @@ trait TestUtils {
...
+  val invalidHead = ""
+
+  val validBase   = "master"
+  val invalidBase = ""
+
+  val validNewPullRequestData   = NewPullRequestData("Amazing new feature", "Please pull this in!")
+  val invalidNewPullRequestData = NewPullRequestData("", "")
+
+  val validNewPullRequestIssue   = NewPullRequestIssue(26)
+  val invalidNewPullRequestIssue = NewPullRequestIssue(5)
```
Comment (fedefernandez) [inline github4s/jvm/src/test/scala/github4s/utils/TestUtils.scala]: The `invalid` vals are not used, could you please remove them?

File: github4s/shared/src/main/scala/github4s/api/PullRequests.scala [hunk 2]
```diff
@@ -80,8 +81,56 @@ class PullRequests[C, M[_]](
...
+   * @param accessToken to identify the authenticated user
+   * @param headers optional user headers to include in the request
+   * @param owner of the repo
+   * @param repo name of the repo
+   * @param newPullRequest	Required. The head and body parameters or the issue parameter
+   * @param head		Required. The name of the branch where your changes are implemented. For cross-repository pull
+   *                             requests in the same network, namespace head with a user like this: username:branch.
+   * @param base		Required. The name of the branch you want the changes pulled into. This should be an existing branch
+   *                             on the current repository. You cannot submit a pull request to one repository that
+   * @param maintainerCanModify		Indicates whether maintainers can modify the pull request.
```
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/api/PullRequests.scala]: Bad indentation, please remove the tabs in the scaladocs. Also, please specify that the value for `maintainerCanModify` is `true` by default

File: github4s/shared/src/main/scala/github4s/api/PullRequests.scala [hunk 3]
```diff
@@ -80,8 +81,56 @@ class PullRequests[C, M[_]](
...
+            accessToken,
+            s"repos/$owner/$repo/pulls",
+            headers,
+            data = CreatePullRequestIssue(
+              newPullRequestIssue.issue,
+              head,
+              base,
+              maintainerCanModify).asJson.noSpaces)
+
+    }
```
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/api/PullRequests.scala]: I think it's more clear to create the `data` first and then make the call, something like:
```scala
def create(
  accessToken: Option[String] = None,
  headers: Map[String, String] = Map(),
  owner: String,
  repo: String,
  newPullRequest: NewPullRequest,
  head: String,
  base: String,
  maintainerCanModify: Option[Boolean] = Some(true)): M[GHResponse[PullRequest]] = {
  val data = newPullRequest match {
    case NewPullRequestData(title, body) ⇒
      CreatePullRequestData(title, head, base, body, maintainerCanModify)
    case NewPullRequestIssue(issue) ⇒
      CreatePullRequestIssue(issue, head, base, maintainerCanModify)
  httpClient.post[PullRequest](accessToken, s"repos/$owner/$repo/pulls", headers, data.asJson.noSpaces)
}
```

**Comment by fedefernandez** [comment]
Also, please add the tests in `[ApiSpec.scala](https://github.com/47deg/github4s/blob/master/github4s/jvm/src/test/scala/github4s/unit/ApiSpec.scala)`

File: github4s/js/src/test/scala/github4s/unit/DecodersSpec.scala [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  "Repository decoder" should "return a valid repo for a valid JSON" in {
+    decode[Repository](getRepoResponse).isRight shouldBe true
+  }
+
+  it should "return an error for an empty JSON" in {
+    decode[Repository](emptyListResponse).isLeft shouldBe true
+  }
+
+  "StatusRepository decoder" should "return a valid repo for a valid JSON" in {
```
Comment (juanpedromoreno) [inline github4s/js/src/test/scala/github4s/unit/DecodersSpec.scala]: CombinedStatus decoder

File: github4s/jvm/src/test/scala/github4s/unit/ApiSpec.scala [hunk 5]
```diff
@@ -608,11 +608,59 @@ class ApiSpec
       accessToken,
       headerUserAgent,
       validRepoOwner,
-      invalidRepoName,
+      validPRRepoName,
```
Comment (juanpedromoreno) [inline github4s/jvm/src/test/scala/github4s/unit/ApiSpec.scala]: Either the test banner or this value are wrong. We should make this consistent.

File: github4s/jvm/src/test/scala/github4s/utils/TestUtils.scala [hunk 6]
```diff
@@ -111,16 +111,33 @@ trait TestUtils {
   val validPullRequestFileSha = "f80f79cafbe3f2ba71311b82e1171e73bd37a470"
   val validPullRequestNumber  = 1
 
+  val validPRRepoName = "sbt-dependencies-test"
```
Comment (juanpedromoreno) [inline github4s/jvm/src/test/scala/github4s/utils/TestUtils.scala]: What would happen if this repository is deleted?

Comment (fedefernandez) [inline github4s/jvm/src/test/scala/github4s/utils/TestUtils.scala]: We could use `github4s`

**Pull request merged**